internal/handlers/jsonb1/where.go

```diff
@@ -146,9 +162,17 @@ func fieldExpr(field string, expr *types.Document, p *pg.Placeholder) (sql strin
 			argSql, arg, err = common.InArray(value.(*types.Array), p, scalar)
 		case "$eq":
 			// {field: {$eq: value}}
-			// TODO special handling for regex
 			sql += "_jsonb->" + p.Next() + " ="
-			argSql, arg, err = scalar(value, p)
+
+			_, isDocument := value.(*types.Document)
+			_, isArray := value.(*types.Array)
+			_, isRegex := value.(types.Regex)
```

Comment:
Regex is not a composite type

Reply:
It just fits the `composite` function so conveniently.

Reply:
Well, does it makes the reading and understanding easier? I would expect a function called `composite` to do something (not sure what without documentation) with a composite type, not with a composite type _and regex_

Reply:
The func was refactored.